DFC.ServiceTaxonomy.ContentApproval/Handlers/ContentApprovalAuditTrailEventHandler.cs

```diff
@@ -0,0 +1,138 @@
+             *  Publish                     - constants.SubmitPublishKey action = submit.Publish                          
+             *  Publish and continue        - constants.SubmitPublishKey action = submit.PublishAndContinue               
+             *  
+             *  Force Publish
+             *      Content Design          - constants.SubmitPublishKey action = submit.RequestApproval - ContentDesign  
+             *      Stakeholder             - constants.SubmitPublishKey action = submit.RequestApproval - Stakeholder    
+             *      SME                     - constants.SubmitPublishKey action = submit.RequestApproval - SME            
+             *      UX                      - constants.SubmitPublishKey action = submit.RequestApproval - UX             
+             *      
+             *  Save Draft and continue     - constants.SubmitSaveKey    action = submit.SaveAndContinue                  
+             *  Save Draft and exit         - constants.SubmitSaveKey    action = submit.Save                             
+             *  
+             *  Request review
+             *      Content Design          - constants.SubmitSaveKey    action = submit.RequestApproval - ContentDesign  
+             *      Stakeholder             - constants.SubmitSaveKey    action = submit.RequestApproval - Stakeholder    
+             *      SME                     - constants.SubmitSaveKey    action = submit.RequestApproval - SME            
+             *      UK                      - constants.SubmitSaveKey    action = submit.RequestApproval - UX             
+             *      
+             *  Send back                   - constants.SubmitSaveKey    action = submit.RequiresRevision                 
+             *               
+             *  Preview draft               - N/A (opens new tab)
+             *  
+             *  Visualise draft graph       - N/A (opens new tab)
+             *  Visualise published graph   - N/A (opens new tab)
+             *  
+             *  Cancel                      - N/A (exits page)
+             */
+            //string actionType = string.Empty;
+
+            if (context is AuditTrailCreateContext<AuditTrailContentEvent> contentEvent)
+            {
+                var contentApprovalPartExists = contentEvent.AuditTrailEventItem.ContentItem.Has<ContentApprovalPart>();
+                if (contentApprovalPartExists)
+                {
+                    var contentApprovalPart = contentEvent.AuditTrailEventItem.ContentItem.As<ContentApprovalPart>();
+                    if (contentApprovalPart != null)
+                    {
+                        var formSubmitAction = contentApprovalPart.FormSubmitAction;
+                        if(formSubmitAction != null)
+                        {
+                            // Publish/Force Publish button
+                            if (formSubmitAction.Item1.Contains(Constants.SubmitPublishKey))
```

Comment:
Could perhaps assign Item1 and Item2 to some variable like button and action in the DisplayDriver?

Reply:
Changed to use (string button, string action) as in the second comment